book/internal-services.md

```diff
@@ -0,0 +1,83 @@
+
+### WATCH
+
+| Custom Resource            | When?                                                      | Why?                                                                               |
+|----------------------------|------------------------------------------------------------|------------------------------------------------------------------------------------|
+| InternalRequest            | Always                                                     | To provide an API to process an internal request                                   |
+| PipelineRun                | Once the PipelineRun is created                            | To relay the Internal PipelineRun status to the remote InternalRequest for viewing |
+
+## Security Risk Mitigations
+
+Enabling remote clusters the ability to run internal jobs carrying certain security risks. The following list describes measures in places to mitigate those risks.
+
+* The creation of an `InternalRequest` custom resource requires permission on the Remote Cluster.
+* The Internal Services controller instance is configured to watch a specific cluster.
+  * This cluster is provided to the controller as an argument to a secret that was added by the admin team.
+  * The secret contains a KUBECONFIG file.
+* Only pipelines that are defined within the controller's namespace can be executed on the Internal, Private cluster.
+  * The deployment of these pipelines is controlled by App Interface and is orchestrated via GitHub pull requests.
+  * In addition, pipelines deployed to production are pinned to Git commit SHAs described in App Interface.
+* The Internal Services controller only acts on remote namespaces that are specifically allowed in the `Config` custom resource.
+
+## Detailed Workflow
+
+1. A `InternalRequest` CR is created by a pipeline run by a service account on the remote cluster as part of a pipeline.
+    * The `spec.request` should reference the pipeline to execute.
+    * The `spec.params` should reference the parameters to pass to the pipeline.
+2. The `InternalRequest` CR is noticed by the Internal Services controller and attempts to reconcile it.
+3. The Internal Services controller verifies whether the remote namespace is allowed in the `Config` CR.
+4. If it is not allowed, the Internal Services controller updates the `InternalRequest` CR `status` field accordingly and stops reconciling.
+5. If it is allowed, a `PipelineRun` is created based on the `spec.request` and `spec.params` from the `InternalRequest` CR.
+6. The `PipelineRun` is watched by the Internal Services controller.
+7. Upon completion, the Internal Services controller updates the `status` section of the `InternalRequest` CR
+8. The remote cluster calling pipeline sees the update to the `status` of the `InternalRequest` CR and continues its execution.
+
+### References
+
+* App Interface - https://gitlab.cee.redhat.com/service/app-interface
```

Comment:
I think this is an implementation detail and isn't needed.

Reply:
removed